schema/msg/msg.go

```diff
@@ -112,6 +112,41 @@ func WritePointMsg(point schema.MetricPoint, buf []byte, version Format) (o []by
 	return nil, fmt.Errorf(errFmtUnsupportedFormat, version)
 }
 
+// WritePointMsgArray is like CreateMsg, except optimized for MetricPoint and buffer re-use.
+// caller must assure a cap-len diff of at least:
```

Comment:
```suggestion
// caller must assure a cap-len diff of exactly:
```

i note that the iterator will try to return errors if there's 1-28 bytes leftover, and try to decode points if there's more bytes than that (but they may not contain a point)

Reply:
If we really want to allow extra space at the end (but I don't think we do?) than we should also encode the number of total elements right after the version

Reply:
I don't think we want to allow extra space. I also want to avoid encoding the number of elements as a prologue as it makes it hard to build the message in a streaming manner.